src/AgentCommerce/SalesChannel/CreateCartRoute.php

```diff
@@ -7,23 +7,111 @@
+        private readonly EntityRepository $productRepository,
+        private readonly CartService $cartService,
+        private readonly PayPalCartTransformer $payPalCartTransformer,
+        private readonly ShopwareCartTransformer $shopwareCartTransformer,
+        private readonly RegisterRoute $registerRoute,
+        private readonly ProductLineItemFactory $lineItemFactory
+    ) {
+    }
+
+    #[Route('/api/paypal/v1/merchant-cart', name: 'api.paypal.merchant-cart', defaults: ['auth_required' => false], methods: [Request::METHOD_POST])]
+    public function createCart(Request $request, SalesChannelContext $salesChannelContext): AgentCartResponse
+    {
+        if (!$productExport = $salesChannelContext->getSalesChannel()->getProductExports()?->first()) {
+            throw AgentException::requiredFieldsMissing('product export');
+        }
+
+        $payPalCart = (new PayPalCart())->assign($request->getPayload()->all());
+
+        if (!$payPalCart->isset('items') || !$payPalCart->getItems()->count()) {
+            throw AgentException::requiredFieldsMissing('items');
+        }
+
+        if ($payPalCart->getCustomer() && !$salesChannelContext->getCustomer()) {
+            $salesChannelContext = $this->loginCustomer($payPalCart, $salesChannelContext);
+        }
+
+        $mapped = [];
+        foreach ($payPalCart->getItems() as $item) {
+            $mapped[$item->getItemId()] = $item->getQuantity();
+        }
+
+        $criteria = new Criteria();
+        $criteria->addFields(['productNumber']);
+        $criteria->addFilter(
+            new EqualsAnyFilter('productNumber', array_keys($mapped)),
+            new EqualsFilter('streams.id', $productExport->getProductStreamId()),
+            new ProductAvailableFilter($salesChannelContext->getSalesChannelId(), ProductVisibilityDefinition::VISIBILITY_LINK)
+        );
+        $products = $this->productRepository->search($criteria, $salesChannelContext->getContext());
+
+        $lineItems = [];
+        foreach ($products as $product) {
+            $lineItems[] = $this->lineItemFactory->create(['id' => $product->getId(), 'quantity' => $mapped[$product->get('productNumber')] ?? 0], $salesChannelContext);
+        }
+
+        if (empty($lineItems)) {
+            throw AgentException::requiredFieldsMissing('items');
+        }
+
+        $swCart = $this->cartService->createNew($salesChannelContext->getToken());
+        $swCart = $this->cartService->add($swCart, $lineItems, $salesChannelContext);
+
+        $createdPayPalCart = $this->payPalCartTransformer->convertToPayPalCart($swCart, $salesChannelContext);
+        $createdPayPalCart->setStatus(PayPalCart::STATUS__CREATED);
+
+        $response = new AgentCartResponse($createdPayPalCart);
+        if ($createdPayPalCart->getStatus() === PayPalCart::STATUS__CREATED) {
+            $response->setStatusCode(Response::HTTP_CREATED);
+        }
+
+        return $response;
```

Comment:
This condition will always be true
```suggestion
        $createdPayPalCart->setStatus(PayPalCart::STATUS__CREATED);

        $response = new AgentCartResponse($createdPayPalCart);
        $response->setStatusCode(Response::HTTP_CREATED);

        return $response;
```

Reply:
The `$createdPayPalCart->setStatus(PayPalCart::STATUS__CREATED);` is not right. Remnants
Needs to be set if the status *is* valid for the creation route. If invalid, the status should be incomplete